src/cov.jl

```diff
@@ -153,3 +153,80 @@ function cor2cov!(C::AbstractMatrix, s::AbstractArray)
     end
     return C
 end
+
+"""
+    CovarianceEstimator
+
+Abstract type for covariance estimators.
+"""
+abstract type CovarianceEstimator end
+
+"""
+    cov(ce::CovarianceEstimator, x::AbstractVector; mean=nothing)
+
+Compute the variance of the vector `x` using estimator `ce`.
+"""
+cov(ce::CovarianceEstimator, x::AbstractVector; mean=nothing) =
+    error("cov is not defined for $(typeof(ce)) and $(typeof(x))")
+
+"""
+    cov(ce::CovarianceEstimator, x::AbstractVector, y::AbstractVector)
+
+Compute the covariance of the vectors `x` and `y` using estimator `ce`.
+"""
+cov(ce::CovarianceEstimator, x::AbstractVector, y::AbstractVector) =
+    error("cov is not defined for $(typeof(ce)), $(typeof(x)) and $(typeof(y))")
+
+"""
+    cov(ce::CovarianceEstimator, X::AbstractMatrix, [w::AbstractWeights]; mean=nothing, dims::Int=1)
+
+Compute the covariance matrix of the matrix `X` along dimension `dims`
+using estimator `ce`. A weighting vector `w` can be specified.
+The keyword `mean` can be
+
+    * `nothing` (default) in which case the mean is estimated then subtracted
+    from the data `X`,
+    * a given number in which case that provided mean is subtracted from the
```

Comment:
I thought I had noted this, but I can't find it: `mean` should be a vector, not a number right? Signatures should be adjusted everywhere. Also, the fallback methods below should accept it.

Above, say "keyword *argument*", and add `:`.

Reply:
Yes, `mean` is not a number but technically it doesn't need to be a `Vector` so maybe it's better to be a bit more vague here.

Reply:
I guess it needs to be an `AbstractVector` at least, right? It can also be useful to mention that it must have a length matching the dimension not specified by `dims`.